src/TypeScriptGenerator.ts

```diff
@@ -429,6 +429,31 @@ function createVisitor(options: TypeGeneratorOptions): IRVisitor.NodeVisitor {
           return [selection];
         });
         state.generatedFragments.add(node.name);
+
+        const refTypeName = getRefTypeName(node.name);
+        const refTypeDataProperty = objectTypeProperty(
+          "$data",
+          ts.createTypeReferenceNode(`${node.name}$data`, undefined)
+        );
+        refTypeDataProperty.questionToken = ts.createToken(
+          ts.SyntaxKind.QuestionToken
+        );
+        const refTypeFragmentRefProperty = objectTypeProperty(
+          "$fragmentRefs",
+          ts.createTypeReferenceNode(
+            getOldFragmentTypeName(node.name),
+            undefined
+          )
+        );
+        const isPluralFragment = isPlural(node);
+        const refType = exactObjectTypeAnnotation([
+          refTypeDataProperty,
+          refTypeFragmentRefProperty
+        ]);
+
+        const dataTypeName = getDataTypeName(node.name);
```

Comment:
Nitpick - but can this line be moved up, and for line 436 to use it also.

Reply:
This issue still exists. If you move this up it can be re-used here too https://github.com/relay-tools/relay-compiler-language-typescript/pull/149/files#diff-152db63141b41dd84d03a80fda984dccR505

Reply:
Done